control-plane/api/v1alpha1/exportedservices_types.go

```diff
@@ -230,14 +234,31 @@ func (in *ExportedService) validate(path *field.Path, consulMeta common.ConsulMe
 }
 
 func (in *ServiceConsumer) validate(path *field.Path, consulMeta common.ConsulMeta) *field.Error {
-	if in.Partition != "" && in.Peer != "" {
-		return field.Invalid(path, *in, "both partition and peer cannot be specified.")
+	count := 0
+
+	if in.Partition != "" {
+		count++
+	}
+	if in.Peer != "" {
+		count++
+	}
+	if in.SamenessGroup != "" {
+		count++
 	}
-	if in.Partition == "" && in.Peer == "" {
-		return field.Invalid(path, *in, "either partition or peer must be specified.")
+	if count > 1 {
+		return field.Invalid(path, *in, "Service consumer must define at most one of Peer, Partition, or SamenessGroup")
+	}
+	if count == 0 {
+		return field.Invalid(path, *in, "Service consumer must define at least one of Peer, Partition, or SamenessGroup")
 	}
 	if !consulMeta.PartitionsEnabled && in.Partition != "" {
-		return field.Invalid(path.Child("partitions"), in.Partition, "Consul Admin Partitions need to be enabled to specify partition.")
+		return field.Invalid(path.Child("partition"), in.Partition, "Consul Admin Partitions need to be enabled to specify partition.")
+	}
+	if in.Partition == WildcardSpecifier {
+		return field.Invalid(path.Child("partition"), "", "exporting to all partitions (wildcard) is not supported")
+	}
+	if in.Peer == WildcardSpecifier {
```

Comment:
Do we support wildcards for SamenessGroup specifications?

Reply:
I did not see it in Consul's validation function, I will ask Eric.

Reply:
Mali discussed with me and Eric, it is not supported so adding a check